### ISSUE TITLE
Hyper Hyper Space: Browser-based peer to peer applications for everything

### ISSUE THREAD

**Issue by sbazerque** [dweb-camp-2019/projects#15]
This project explores the idea of using  web browsers (and in the future mobile devices) as nodes in a peer to peer network that provides end-user platforms (social networking, communications, commerce, sharing, etc).

The main advantages of using the browser as a platform are:

- low friction for users, since there's no software to install
- stronger democratizing force, since technologist and non-technologists can both run network nodes

We're close to having a working implementation of a chat application, with other solutions coming after that (commerce, sharing, etc).

There's a bit more information in https://www.hyperhyperspace.org

I think it would be interesting to discuss these ideas in the camp :-)

**Comment by benhylau** [comment]
>using the browser as a platform

I agree! This is related to my personal work as well, workshops where people only need a laptop or phone without having to install anything. Curious what types of discussions you may want to explore, or perhaps hands-on sessions?

I usually run the software on Raspberry Pis that are meshed together. There will be switches where you can plug Raspberry Pis or other devices into the local network if you want.

**Comment by sbazerque** [comment]
I think there are several directions that are worth exploring, elaborating on the idea of having a platform that enables users to have all their information stored within their own devices and running a  software platform collaboratively using just their standards-compliant browsers:

1) Finding collaboration ideas that have no obvious business model, or that could be improved by a non-for-profit approach, but that could be implemented within this platform (e.g. sharing knowledge or help, lending tools, books or other physical stuff, the de-intermediation ideas sketched on our website).

2) How business (new or existing) could create value in novel ways by integrating with a platform that is more open and where users have more control over their data (both to withdraw the personal bits, and to share what they need to get out: commerce, sharing, etc)

3) Synergies with other peer-to-peer technologies.

About the Pis, the only bit that needs to run outside of the browser is the signalling software, which is just a bit of Python. It could run on someone's laptop also.

There's also the problem of delivering the static site to people's browsers, that could be done with a Pi but maybe it's overkill? Also IPFS, maybe?

I hope to publish the platform prototype soon. I fear all this may sound a bit looney otherwise :)

**Comment by benhylau** [comment]
Hey how's this coming along? Please submit https://docs.google.com/forms/d/1Iet2xk6qwZZWkDHTEMCUn5zlb80IwXNtb6epk4-9O-s/ if you would like to list this project on website and schedule :D

**Comment by sbazerque** [comment]
Hopefully there'll be a beta to play along at the camp :-)

Thanks for the link, I'll fill a proposal today!

**Comment by amirouche** [comment]
Very well described. I hope dweb camp conversation will continue after the camp ends.

**Comment by benhylau** [comment]
@amirouche responded to you on new thread!

**Comment by sbazerque** [comment]
@benhylau just sent the proposal, back to programming now 👍
